api/server/server.go

```diff
@@ -847,12 +847,28 @@ func postContainersRestart(eng *engine.Engine, version version.Version, w http.R
 	if vars == nil {
 		return fmt.Errorf("Missing parameter")
 	}
-	job := eng.Job("restart", vars["name"])
-	job.Setenv("t", r.Form.Get("t"))
-	if err := job.Run(); err != nil {
+
+	seconds := 10
+	t := r.Form.Get("t")
+	if t != "" {
+		s, err := strconv.Atoi(t)
+		if err != nil {
+			return err
+		}
+		seconds = s
+	}
+	//s, err := strconv.Atoi(t)
```

Comment:
Hmmm, looks like it was 0 if `t` is not passed in querystring, because it is always exists in env, even if it is `""`.